# Bonus: Temperature Analysis I

In [1]:
import pandas as pd
from datetime import datetime as dt

In [2]:
# "tobs" is "temperature observations"
df = pd.read_csv('Resources/hawaii_measurements.csv')
df.head()

station        date  prcp  tobs
0  USC00519397  2010-01-01  0.08    65
1  USC00519397  2010-01-02  0.00    63
2  USC00519397  2010-01-03  0.00    74
3  USC00519397  2010-01-04  0.00    76
4  USC00519397  2010-01-06   NaN    73

In [3]:
# Convert the date column format from string to datetime
df['date'] = pd.to_datetime(df['date'])
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19550 entries, 0 to 19549
Data columns (total 4 columns):
 #   Column   Non-Null Count  Dtype         
---  ------   --------------  -----         
 0   station  19550 non-null  object        
 1   date     19550 non-null  datetime64[ns]
 2   prcp     18103 non-null  float64       
 3   tobs     19550 non-null  int64         
dtypes: datetime64[ns](1), float64(1), int64(1), object(1)
memory usage: 611.1+ KB


In [4]:
# Set the date column as the DataFrame index
df = df.set_index(df['date'])
df.head()

station       date  prcp  tobs
date                                          
2010-01-01  USC00519397 2010-01-01  0.08    65
2010-01-02  USC00519397 2010-01-02  0.00    63
2010-01-03  USC00519397 2010-01-03  0.00    74
2010-01-04  USC00519397 2010-01-04  0.00    76
2010-01-06  USC00519397 2010-01-06   NaN    73

### Compare June and December data across all years 

In [5]:
from scipy import stats

In [6]:
# Filter data for desired months
june_df = df[df.index.month == 6]
june_df.head()

station       date  prcp  tobs
date                                          
2010-06-01  USC00519397 2010-06-01  0.00    78
2010-06-02  USC00519397 2010-06-02  0.01    76
2010-06-03  USC00519397 2010-06-03  0.00    78
2010-06-04  USC00519397 2010-06-04  0.00    76
2010-06-05  USC00519397 2010-06-05  0.00    77

In [7]:
dec_df = df[df.index.month == 12]
dec_df.head()

station       date  prcp  tobs
date                                          
2010-12-01  USC00519397 2010-12-01  0.04    76
2010-12-03  USC00519397 2010-12-03  0.00    74
2010-12-04  USC00519397 2010-12-04  0.00    74
2010-12-06  USC00519397 2010-12-06  0.00    64
2010-12-07  USC00519397 2010-12-07  0.00    64

In [8]:
# Identify the average temperature for June
june_df2 = june_df.mean()
june_df2

<ipython-input-8-e319cd336b17>:2: FutureWarning: DataFrame.mean and DataFrame.median with numeric_only=None will include datetime64 and datetime64tz columns in a future version.
  june_df2 = june_df.mean()


prcp     0.136360
tobs    74.944118
dtype: float64

In [9]:
# Identify the average temperature for December
dec_df2 = dec_df.mean()
dec_df2

<ipython-input-9-88d691d2e017>:2: FutureWarning: DataFrame.mean and DataFrame.median with numeric_only=None will include datetime64 and datetime64tz columns in a future version.
  dec_df2 = dec_df.mean()


prcp     0.216819
tobs    71.041529
dtype: float64

In [10]:
# Create collections of temperature data
june_tobs = june_df.tobs

dec_tobs = dec_df.tobs

print(june_tobs, dec_tobs)

date
2010-06-01    78
2010-06-02    76
2010-06-03    78
2010-06-04    76
2010-06-05    77
              ..
2017-06-26    79
2017-06-27    74
2017-06-28    74
2017-06-29    76
2017-06-30    75
Name: tobs, Length: 1700, dtype: int64 date
2010-12-01    76
2010-12-03    74
2010-12-04    74
2010-12-06    64
2010-12-07    64
              ..
2016-12-27    71
2016-12-28    71
2016-12-29    69
2016-12-30    65
2016-12-31    65
Name: tobs, Length: 1517, dtype: int64


In [11]:
# Run unpaired t-test to compare the means of the temperature in June and December.
stats.ttest_rel(june_df2, dec_df2)

Ttest_relResult(statistic=0.9595993181076836, pvalue=0.5131232081478694)

### Analysis

The unpaired t-test anaylsis above was peformed to compare two averages (means) between temperatures in Hawaii in the months of June and December with the goal to learn if the difference between the two averages were zero. The unpaired t-test was used because we are comapring two independent, unrelated times in a year (June and December). The results as displayed were: test statistic was 0.96 and the corresponding two sided p-value was 0.51. Becuase the p-value is more than 0.05, we have sufficient evidence to say that the mean temperature values between June and December in Hawaii are not statistically significant.  